security/pkg/k8s/controller/workloadsecret_test.go

```diff
@@ -357,76 +380,135 @@ func TestUpdateSecret(t *testing.T) {
 	}
 }
 
-func TestSecretOptIn(t *testing.T) {
+func TestManagedNamespaceRules(t *testing.T) {
+	testCases := map[string]struct {
+		ns                        *v1.Namespace
+		selfNamespace             string
+		enableNamespacesByDefault bool
+		result                    bool
+	}{
+		"not managed by default, no override, and namespace label does not match actual ns => no secret": {
+			ns:                        createNS("unlabeled", map[string]string{}),
+			selfNamespace:             "random",
+			enableNamespacesByDefault: false,
+			result:                    false,
+		},
+		"not managed by default, no override, and namespace matches => secret": {
+			ns:                        createNS("unlabeled", map[string]string{NamespaceManagedLabel: "test-ns"}),
+			selfNamespace:             "test-ns",
+			enableNamespacesByDefault: false,
+			result:                    true,
+		},
+		"not managed by default, override is false, and namespace matches => no secret": {
+			ns:                        createNS("unlabeled", map[string]string{NamespaceManagedLabel: "test-ns", NamespaceOverrideLabel: "false"}),
+			selfNamespace:             "test-ns",
+			enableNamespacesByDefault: false,
+			result:                    false,
+		},
+		"is managed by default, override is not present, and no namespace tag => secret": {
+			ns:                        createNS("unlabeled", map[string]string{}),
+			selfNamespace:             "test-ns",
+			enableNamespacesByDefault: true,
+			result:                    true,
+		},
+		"is managed by default, override is false, and no namespace tag => no secret": {
+			ns:                        createNS("unlabeled", map[string]string{NamespaceOverrideLabel: "false"}),
+			selfNamespace:             "test-ns",
+			enableNamespacesByDefault: true,
+			result:                    false,
+		},
+	}
+
+	for k, tc := range testCases {
+		client := fake.NewSimpleClientset()
```

Comment:
you can do t.Run(k, func ... ) and then you can distinguish individual test failures or run just one test case, etc. Makes things a bit easier to understand https://blog.golang.org/subtests

Reply:
changed to use `t.Run` 👍